### ISSUE TITLE
I get the same error as in #1617 #11136 on MPS

### ISSUE THREAD

**Issue by BetsaDmytro** [ultralytics/yolov5#13704]
### Search before asking

- [x] I have searched https://github.com/ultralytics/yolov5/issues and did not find a similar report.

### Project area

Other, Training

### Bug

issue is happening due to this lines
https://github.com/ultralytics/yolov5/blob/85f6019e5af2641e33139e97415b7bd1dc72d779/utils/general.py#L913-L915

The same error as in https://github.com/ultralytics/yolov5/issues/1617 and in https://github.com/ultralytics/yolov5/issues/11136 occurs.

### Environment

    YOLO: latest as of now
    OS: MacOS 13.2.1

### Minimal Reproducible Example

```
# load PIL images into list v

device = "mps"
model = torch.hub.load("ultralytics/yolov5", "yolov5s")
model.to(device);
model(v)
```

### Additional

I am confident that together we can finally resolve this monstrosity of an issue and remove one line of code

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @BetsaDmytro, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).  
This is an automated response 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us debug it. You included a great start—please expand it into a fully runnable MRE that includes the exact install steps/commit, the full traceback, and a tiny self-contained sample input (or a script that generates one) so we can reproduce on MPS reliably 🙏  
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>  
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help the team quickly triage your MPS report 🧪, please also include in the MRE:
-Exact YOLOv5 source (commit hash or whether it’s `torch.hub.load()` default) and whether you’re running from a fresh clone
-Exact PyTorch version and whether MPS is enabled
-Full error log/traceback output (not just the referenced lines)
-Whether this reproduces with the provided [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py) entrypoint or only via hub-loading

**Comment by pderrenger** [comment]
That `prediction = prediction.cpu()` guard exists because Apple MPS support is still incomplete and NMS frequently has to run on CPU, so removing it usually just changes where the failure happens; could you paste the full traceback and the output of `python -c "import torch, torchvision; print(torch.__version__, torchvision.__version__)"`, and confirm it still reproduces on the current `master` with a fresh Hub pull (`torch.hub.load("ultralytics/yolov5", "yolov5s", force_reload=True)`) ? For a practical workaround meanwhile, either run on CPU or load with `autoshape=False` (and pass a preprocessed tensor) to avoid the AutoShape postprocessing path that can hit device-mismatch edge cases on MPS; for context on the current MPS status see the [YOLOv5 v6.2 release notes](https://www.ultralytics.com/blog/simplifying-classification-workflows-with-yolov5-v6-2).
